### ISSUE TITLE
TaskDens inheritance

### ISSUE THREAD

**Issue by RaphaelS1** [mlr-org/mlr3proba#157]
Due to updates in mlr3 we now have to make a decision about if TaskDens should inherit from TaskSupervised or TaskUnsupervised.

Strictly speaking it should be `TaskUnsupervised` however this means there is no 'truth' column in the Prediction object and therefore all `MeasureDens` object must have the 'requires_task' and 'requires_train_set' properties and a user will have to manually pass in the training set every time they want to score a density learner (note this is automatic in benchmark).

Using `TaskSupervised` allows there to be a 'truth' column but of course this won't strictly speaking be the actual truth.

So in short, `TaskSupervised` is simpler for programmers and users but `TaskUnsupervised` is mathematically correct.

@fkiraly @aintoha
